file/sequence_file_reader.cc

```diff
@@ -25,19 +26,32 @@ namespace ROCKSDB_NAMESPACE {
 IOStatus SequentialFileReader::Create(
     const std::shared_ptr<FileSystem>& fs, const std::string& fname,
     const FileOptions& file_opts, std::unique_ptr<SequentialFileReader>* reader,
-    IODebugContext* dbg) {
+    IODebugContext* dbg, RateLimiter* rate_limiter) {
   std::unique_ptr<FSSequentialFile> file;
   IOStatus io_s = fs->NewSequentialFile(fname, file_opts, &file, dbg);
   if (io_s.ok()) {
-    reader->reset(new SequentialFileReader(std::move(file), fname));
+    reader->reset(new SequentialFileReader(std::move(file), fname, nullptr, {},
+                                           rate_limiter));
   }
   return io_s;
 }
 
-IOStatus SequentialFileReader::Read(size_t n, Slice* result, char* scratch) {
+IOStatus SequentialFileReader::Read(size_t n, Slice* result, char* scratch,
```

Comment:
cc @anand1976  our file read wrapper expert - would you mind also taking a quick look into this PR's change on `SequentialFileReader::Read`.  Although I checked against its original impl as well as an example `RandomAccessFileReader::Read` PLUS [ReadaheadSequentialFileTest](https://github.com/facebook/rocksdb/blob/dd203ed6041000ba728a7ad92b3d568ccd4d4e21/util/file_reader_writer_test.cc#L586) seems to be a good testing coverage for (SequentialFileReader + no rate limiter) case, I'd still appreciate a quick look from domain expert. Thanks!

Reply:
Reached out offline.